### PR TITLE
Add acorn update --pull functionality

### PR DESCRIPTION
**Pull request by cjellick**
In adding the auto-upgrade feature, we lost the abilty to force a
re-pull of an image for an app. This adds that ability back by
explicitly adding the ability to do `acorn update --pull ...`

Addresses #948

Signed-off-by: Craig Jellick <craig@acorn.io>

### CODE REVIEW

File: pkg/controller/appdefinition/pullappimage.go [hunk 1]
```diff
@@ -83,8 +83,12 @@ func determineTargetImage(appInstance *v1.AppInstance) (string, string) {
...
-		// Auto-upgrade is off. Only need to pull if spec and status are not equal
+		// Auto-upgrade is off. Only need to pull if spec and status are not equal or we're trying to trigger a repull
 		if appInstance.Spec.Image == appInstance.Status.AppImage.Name {
+			// Repull triggered
+			if appInstance.Status.AvailableAppImage == appInstance.Spec.Image {
+				return appInstance.Spec.Image, ""
+			}
 			return "", ""
 		} else {
 			return appInstance.Spec.Image, ""
```
Comment (tylerslaton) [inline pkg/controller/appdefinition/pullappimage.go:94]: non-blocking: We can flip the if statement here to keep the code a bit dryer and more fitting for the comment.
```suggestion
		// Auto-upgrade is off. Only need to pull if spec and status are not equal or we're trying to trigger a repull
		if appInstance.Spec.Image != appInstance.Status.AppImage.Name || 
		appInstance.Status.AvailableAppImage == appInstance.Spec.Image {
		        return appInstance.Spec.Image, ""
		} else {
		        return "", ""
```

File: pkg/client/client.go [hunk 2]
```diff
@@ -109,8 +109,6 @@ type AppUpdateOptions struct {
 
 type LogOptions apiv1.LogOptions
 
-type ConfirmUPgrade apiv1.ConfirmUpgrade
```
Comment (tylerslaton) [inline pkg/client/client.go:112]: non-blocking: Weird that this exists, I guess its just some left-over dead code from the original auto-upgrade feature?

**Review by tylerslaton — APPROVED** [review]
Looks good to me. Nothing blocking, one question and one suggestion if you've got the time.

File: pkg/client/client.go (hunk 2 above)
Comment (cjellick) [reply pkg/client/client.go:112]: Yeah. I had added it when I was basically copying the pattern for the feature from previous features, but TBH, I never knew what it did and the PR was so big that I didn't catch it during cleanup.

**Review by jsilverio22 — APPROVED** [review]
LGTM

File: pkg/controller/appdefinition/pullappimage.go (hunk 1 above)
Comment (cjellick) [reply pkg/controller/appdefinition/pullappimage.go:94]: makes sense. thanks!

**Pull request merged**